src/sctools/bam.py

```diff
@@ -441,6 +441,42 @@ def get_tag_or_default(alignment: pysam.AlignedSegment, tag_key: str, default: O
         return default
 
 
+def sort_by_tags_and_queryname(
+        records: Iterable[pysam.AlignedSegment],
+        tags: Iterable[str] = None) -> Iterable[pysam.AlignedSegment]:
+    """Sorts the given bam records by the given tags, followed by query name.
+    If no tags are given, just sorts by query name.
+    """
+    if tags is None:
+        tags = []
+    comparable_records = []
+    for r in records:
+        cr = TagSortableRecord.from_aligned_segment(r, tags)
+        comparable_records.append(cr)
+    sorted_records = sorted(comparable_records)
+    return TagSortableRecord.get_aligned_segments(sorted_records)
+
+
+def read_bam(input_path: str) -> (pysam.AlignmentFile.header, List[pysam.AlignedSegment]):
```

Comment:
Interesting, does mypy validate explicit tuples? My experience is that this should be `Tuple[pysam.AlignmentFile.header, List[pysam.AlignedSegment]]`

Reply:
You're right, it wants it the way you said. Will fix.